app/controllers/api/v1/statuses/reblogs_controller.rb

```diff
@@ -14,12 +14,17 @@ def create
   end
 
   def destroy
-    @status = status_for_destroy.reblog
-    @reblogs_map = { @status.id => false }
+    begin
+      @status = status_for_destroy.reblog
+    rescue ActiveRecord::RecordNotFound
+      @status = Status.find(params[:status_id])
+    else
```

Comment:
Also you can't chain `rescue` with `else` unless this is some super wild syntax thing I never encountered

Reply:
I am pretty sure you can :thinking:

> The code in an else clause is executed if the code in the body of the begin statement runs to completion without exceptions. If an exception occurs, then the else clause will obviously not be executed. The use of an else clause is not particularly common in Ruby.
http://rubylearning.com/satishtalim/ruby_exceptions.html